### PR TITLE
Improve error handling and logging when using a remote cache

### PR DESCRIPTION
**Pull request by henryaj**
- Print stacktrace in debug logs on panic
- Fail fast if remote execution is enabled but remote end doesn't support it
- Gracefully handle invalid responses from build server (e.g. when expected proto fields are left nil)
- Print helpful error if remote build exits with non-zero code

In time I'd like to improve all this such that remote build failures are non-fatal - please should just fall back to building locally I guess.

### CODE REVIEW

File: src/cache/dir_cache.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Diretory-based cache.
+// Directory-based cache.
```
Comment (peterebden) [inline src/cache/dir_cache.go]: much shame:(

File: src/remote/blobs.go [hunk 2]
```diff
@@ -311,6 +311,9 @@ func (c *Client) downloadBlobs(f func(ch chan<- *blob) error) error {
 
 // retrieveByteStream receives a file back from the server as a byte stream.
 func (c *Client) retrieveByteStream(b *blob) error {
+	if b.Digest == nil {
+		return fmt.Errorf("can't retrieve byte stream from nil digest")
```
Comment (peterebden) [inline src/remote/blobs.go]: ooh dear does this happen? it probably shouldn't...

Comment (henryaj) [reply src/remote/blobs.go]: Maybe - it's to do with reading [these digest fields](https://github.com/bazelbuild/remote-apis/blob/master/build/bazel/remote/execution/v2/remote_execution.proto#L870-L885) which I guess are optional, strictly speaking.

File: src/cache/dir_cache.go (hunk 1 above)
Comment (henryaj) [reply src/cache/dir_cache.go]: :smug:

**Pull request merged**